pinecone/client.go

```diff
@@ -80,14 +207,116 @@ func NewClientBase(in NewClientBaseParams) (*Client, error) {
 	return &c, nil
 }
 
-func (c *Client) Index(host string) (*IndexConnection, error) {
+// Index creates an IndexConnection to the specified host.
+//
+// Parameters:
+//   - host: The host URL of your Pinecone index.
+//
+// Returns a pointer to an IndexConnection instance on success. In case of failure, it returns nil and an error.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//    ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//    SourceTag: "your_source_identifier", // optional
+//   }
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//   }
```

Comment:
nit: Is it just me or are these closing brackets indented one space? I keep thinking I'm seeing indentation problems but it's hard to tell.